samza-test/src/main/java/org/apache/samza/test/framework/TestRunner.java

```diff
@@ -262,6 +275,14 @@ public void run(Duration timeout) {
     if (status.getStatusCode() == ApplicationStatus.StatusCode.UnsuccessfulFinish) {
       throw new SamzaException(ExceptionUtils.getStackTrace(status.getThrowable()));
     }
+    // Clean the state directory if any
+    Path stateDir = Paths.get(configs.get(JobConfig.JOB_NON_LOGGED_STORE_BASE_DIR()));
+    if (!Files.exists(stateDir)) return;
+    try {
+      Files.walk(stateDir).sorted(Comparator.reverseOrder()).map(Path::toFile).forEach(File::delete);
```

Comment:
There's an existing utility method `FileUtil.rm(file)` in `FileUtil.scala` in samza which recursively deletes the entire directory. This duplication here can be removed by simply switching to it.